edge/services/helloSecretWorld/CreateService.md

```diff
@@ -0,0 +1,262 @@
+    - Note: this service is a shell script simply for brevity, but you can write your service in any language.
+
+6. Build the service docker image.
+
+  ```bash
+  make
+  ```
+
+7. Edit the secret value in `horizon/hw_who` to whatever you want. The secret key is ignored by the code in `service.sh`, but the code will read the secret value from the file and output it using the `echo` statement. Because we will be testing our service with `hzn dev service start` without a deployment pattern or policy, the agent cannot get the secret details of any secrets in the secrets manager. Instead, it can read this file as the secret details of the secret used by the service. 
+
+8. Test the service by running it the simulated agent environment. Pass in the **absolute path** to the `hw_who` file, which is in the `horizon` subdirectory. 
+
+  ```bash
+  hzn dev service start --secret=<abs-path-to-hw_who>
+  ```
+
+Upon starting the service, this command will mount the local `hw_who` file containing the contents of the secret to the service container in the `open-horizon-secrets` folder. If this file is changed while the service is running, the change will be reflected in the container as well.
+
+8. Check that the container is running:
+
+  ```bash
+  sudo docker ps
+  ```
+
+9. Display the environment variables Horizon passes into your service container. Note the variables that start with `HZN_ESS_`. These are used by the service to query the secrets API for updating secrets.
+
+  ```bash
+  sudo docker inspect $(sudo docker ps -q --filter name=$SERVICE_NAME) | jq '.[0].Config.Env'
+  ```
+
+10. View the service output (you should see messages like **\<your-node-id\> says: Hello \<secret-value\>!**, where \
+\<secret-value\> is the value of the secret as stated in the `hw_who` file.
+
+  on **Linux**:
+
+  ```bash
+  eval $(hzn util configconv -f horizon/hzn.json)
+  sudo tail -f /var/log/syslog | grep ${SERVICE_NAME}[[]
+  ```
+
+  on **Mac**:
+
+  ```bash
+  eval $(hzn util configconv -f horizon/hzn.json)
+  sudo docker logs -f $(sudo docker ps -q --filter name=$SERVICE_NAME)
+  ```
+
+11. Stop the service:
+
+  ```bash
+  hzn dev service stop
+  ```
+
+12. You are now ready to publish your edge service and pattern, so that they can be deployed to real edge nodes. Instruct Horizon to push your docker image to your registry and publish your service in the Horizon Exchange:
+
+  ```bash
+  hzn exchange service publish -f horizon/service.definition.json
+  hzn exchange service list
+  ```
+
+13. (This step must be done by an organization admin) Create an organization-wide secret called `secret-name` in your organization's secrets manager. The flags `--secretKey` and `--secretDetail` provide the key and value for the secret, respectively. The key is ignored by the service, but `"Hello <secret-value>!"` will be printed by the service.
```

Comment:
I think we should delete the last sentence, they may have modified service.sh to not do this anymore.